src/main/kotlin/nmcp/NmcpExtension.kt

```diff
@@ -1,28 +1,42 @@
 package nmcp
 
+import gratatouille.GExtension
+import gratatouille.capitalizeFirstLetter
+import nmcp.internal.configureAttributes
+import nmcp.internal.nmcpProducerConfigurationName
+import nmcp.internal.task.registerPublishTask
+import nmcp.internal.withRequiredPlugin
 import org.gradle.api.Action
 import org.gradle.api.Project
 import org.gradle.api.publish.PublishingExtension
 import org.gradle.api.tasks.bundling.Zip
-import org.gradle.configurationcache.extensions.capitalized
 
+@GExtension(pluginId = "com.gradleup.nmcp")
 open class NmcpExtension(private val project: Project) {
-    private var mavenPublishFound = false
+    internal val spec = project.objects.newInstance(CentralPortalOptions::class.java)
+    // Lifecycle task to publish all the publications in the given project
     private val publishAllPublicationsToCentralPortal = project.tasks.register("publishAllPublicationsToCentralPortal")
 
-    private fun register(publicationName: String?, spec: NmcpSpec) {
-        if (publicationName != null) {
-            registerInternal(publicationName, spec)
-        } else {
+    init {
+        project.configurations.create(nmcpProducerConfigurationName) {
+            it.isCanBeConsumed = true
+            it.isCanBeResolved = false
+            // See https://github.com/GradleUp/nmcp/issues/2
+            it.isVisible = false
+
+            it.configureAttributes(project)
+        }
+
+        project.withRequiredPlugin("maven-publish", {
             val publishing = project.extensions.findByType(PublishingExtension::class.java)!!
             publishing.publications.configureEach {
-                registerInternal(it.name, spec)
+                registerInternal(it.name)
             }
-        }
+        })
```

Comment:
```suggestion
        }
```

+ correct `project.withRequiredPlugin("maven-publish", {` to `project.withRequiredPlugin("maven-publish") {` as per [Kotlin style guide](https://kotlinlang.org/docs/coding-conventions.html#lambdas) for lambdas